packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx

```diff
@@ -199,13 +203,148 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
         });
     }
 
+    /**
+     * @param filePath Path of the file.
+     * @param rootUris URIs of the root folders in the current workspace.
+     * @returns URI of the root folder that a file is belong to.
+     * If a file belongs to more than one root folder, return the root folder closest to the file.
+     * If the file is not from the current workspace, return an empty URI.
+     */
+    getRoot(filePath: string, rootUris: string[]): URI {
+        const roots = rootUris.filter(root => new URI(root).withScheme('file').isEqualOrParent(new URI(filePath).withScheme('file')));
+        if (roots.length > 0) {
+            return new URI(roots.sort((r1, r2) => r2.length - r1.length)[0]);
+        }
+        return new URI();
+    }
+
+    /**
+     * Find matches in a single widget.
+     * @param searchTerm  The search keyword.
+     * @param widget The widget that is being searched.
+     * @param fileContent The content of the widget.
+     * @param searchOptions The options for search operation.
+     */
+    findMatches(searchTerm: string, widget: EditorWidget, fileContent: string, searchOptions: SearchInWorkspaceOptions): SearchMatch[] {
+        let matches: SearchMatch[] = [];
+        const allLines: string[] = fileContent.split(/[\n\u0085\u2028\u2029]|\r\n?/);
```

Comment:
We can replace this method with the API from `monaco`:

- [getLineContent](https://microsoft.github.io/monaco-editor/api/interfaces/monaco.editor.itextmodel.html#getlinecontent)
- [getLinesContent](https://microsoft.github.io/monaco-editor/api/interfaces/monaco.editor.itextmodel.html#getlinescontent)